### ISSUE TITLE
debug logging needs to be beefed up significantly

### ISSUE THREAD

**Issue by nnordrum** [uber/makisu#120]
Assume you don't know what you're doing, or there are problems that you go "arg! I wish they had better logging!"

Do that :)

**Comment by yiranwang52** [comment]
I agree, but do you have some specific examples to help us understand the pain points?

**Comment by nnordrum** [comment]
A non-inclusive list:
* each command line parameter as you resolve it
* state after you resolve all parameters
* mechanism to see the HTTP req/resp
* before, during, and after template variable resolution

I think that will keep you busy and solve _most_ problems. Basically, it’s difficult to fix things that you’re doing wrong. Works great though when all is right though :)
